CtaReactiveMaster/Sources/Home/HomeViewController.swift

```diff
@@ -5,11 +5,103 @@
+        didSet{
+            tableView.dataSource = self
+            tableView.delegate = self
+            tableView.register(NewsTableViewCell.nib, forCellReuseIdentifier: NewsTableViewCell.identifier)
+            tableView.refreshControl = UIRefreshControl()
+        }
+    }
+    
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    
+    fileprivate var articles:[Article] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        navigationItem.title = "NewsAPI"
+        navigationController?.navigationBar.titleTextAttributes
+            = [NSAttributedString.Key.font: UIFont(name: "Times New Roman", size: 24)!]
+        navigationController?.navigationBar.backgroundColor = UIColor.brown
+        
+        activityIndicator.hidesWhenStopped = true
+        tableView.refreshControl?.addTarget(self, action: #selector(refresh(sender:)), for: .valueChanged)
+       
+        activityIndicator.startAnimating()
+        fetchNewsAPI()
+        activityIndicator.stopAnimating()
+    }
+    
+    private func fetchNewsAPI(){
+        
+        AF.request("http://newsapi.org/v2/everything?q=bitcoin&from=2020-12-08&sortBy=publishedAt&apiKey=67945148525042b9b63954def7a50c38").response { [weak self] response in
+
+            switch response.result {
+                case .success(let data):
+                    guard let unwrappedData = data
+                    else{
+                        return
+                    }
+                    
+                    do{
+                        //APIのデータに従った構造体を書く → 取得するJSONのデータに合わせた構造体jsonDataを定義
+                        let jsonData = try JSONDecoder().decode(JsonData.self, from: unwrappedData)
+                        self?.articles = jsonData.articles
+                    }
+                    catch let error{
+                        print(error)
+                        return
+                    }
+                    self?.tableView.reloadData()
+                    break
+                    
+                case .failure(let error):
+                    UIAlertController.showRetryAlert(to: self, with: error, retryhandler: self?.fetchNewsAPI ?? {()-> Void in return})
+                    print(error)
+            }
+            
+            let isRefreshing = self?.tableView.refreshControl?.isRefreshing ?? false
+            if isRefreshing{
+                self?.tableView.refreshControl?.endRefreshing()
+            }
+        }
+    }
+    
+    @objc func refresh(sender: UIRefreshControl){
+        fetchNewsAPI()
+    }
+}
+
+extension HomeViewController : UITableViewDelegate{
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let url = articles[indexPath.row].url
+        else {
+            return
+        }
```

Comment:
ditto

Reply:
92行目のurlで使います！

Reply:
改行の仕方の話

Reply:
```suggestion
        guard let url = articles[indexPath.row].url else { return }
```
```suggestion
        guard let url = articles[indexPath.row].url else { 
             return 
         }
```
どっちかに修正してほしい。他のguard文使ってる箇所も。